src/development/submodules.md

```diff
@@ -43,3 +43,33 @@ E: Error validating submodules in tree:
 ```
 
 Since the Platform.sh Git server cannot connect to Github via SSH without being granted an SSH key to do so, you should not be using an SSH URL: ``git@github.com:...``, but you should use an HTTPS URL instead: ``https://github.com/...``.
+
+## Use of private git repositories
+
+When using git submodules that are hosted on private repositories, using the `https` protocol will fail with errors like:
+
+```
+GitProtocolError: unexpected http resp 401 for https://bitbucket.org/myusername/mymodule.git/info/refs?service=git-upload-pack
+```
+
+To fix this, you need:
+
+1. To move your `.gitmodules` file from the https syntax to the ssh syntax, e.g.
+
+from:
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url = https://bitbucket.org/myusername/mymodule.git
+```
+
+to
+
+```
+[submodule "support/mymodule"]
+    path = support/mymodule
+    url=git@bitbucket.org:myusername/mymodule.git
+```
+
+2. to add the ssh public key in the platform.sh project settings "Deploy Key" tab in our Web UI as per the [Private Repository](development/private-repository.md) documentation page, to allow our git service to pull the module from the remote git service. This assumes of course you have configured the remote git repository to allow this by generating a private/public key pair, see this [bitbucket](https://confluence.atlassian.com/bitbucket/use-ssh-keys-in-bitbucket-pipelines-847452940.html) page for instance.
```

Comment:
* "To add", capital T.
* SSH, not ssh.  (Here and above.)
* Git, not git. It's a product name.  (Here and above.)
* Platform.sh.  Capitalize us. :-)
* Bitbucket. Capitalize them.
* "of course" is unnecessary, remove it.
* ", to allow" should be ", which will allow".
* "key pair, see this" is a run-in.  It should be "key pair. For example, see the [Bitbucket documentation]".